test/ConfigurationFactoryTest.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace RoaveTest\PsrContainerDoctrine;
+
+use Doctrine\ORM\Cache\CacheConfiguration;
+use Doctrine\ORM\Cache\DefaultCacheFactory;
+use Doctrine\Persistence\Mapping\Driver\MappingDriver;
+use PHPUnit\Framework\TestCase;
+use Psr\Cache\CacheItemPoolInterface;
+use Psr\Container\ContainerInterface;
+use ReflectionProperty;
+use Roave\PsrContainerDoctrine\ConfigurationFactory;
+
+use const PHP_VERSION_ID;
+
+final class ConfigurationFactoryTest extends TestCase
+{
+    public function testWillSetCacheItemPoolCaches(): void
+    {
+        $resultCache    = $this->createMock(CacheItemPoolInterface::class);
+        $queryCache     = $this->createMock(CacheItemPoolInterface::class);
+        $metadataCache  = $this->createMock(CacheItemPoolInterface::class);
+        $hydrationCache = $this->createMock(CacheItemPoolInterface::class);
+
+        $container = $this->createMock(ContainerInterface::class);
+        $config    = [
+            'doctrine' => [
+                'configuration' => [
+                    'orm_default' => [
+                        'metadata_cache' => 'metadata',
+                        'result_cache' => 'result',
+                        'query_cache' => 'query',
+                        'hydration_cache' => 'hydration',
+                        'second_level_cache' => ['enabled' => true],
```

Comment:
This config actually led to the bugfix mentioned in https://github.com/Roave/psr-container-doctrine/pull/63/files#r806931084